tools/bouncer_worker/src/lib/processMonitor.js

```diff
@@ -51,17 +52,19 @@ const updateMemory = async (pid) => {
 			dataByPid[pid].maxMemory = Math.max(dataByPid[pid].maxMemory, data);
 		}
 	} catch (err) {
-		logger.error(`[maxmem]: ${err}`, logLabel);
+		if (dataByPid[pid]) {
+			logger.error(`[ProcessMonitor.updateMemory]: ${err}`, logLabel);
+		}
 	}
 };
 
 const monitor = (pid) => setInterval(() => updateMemory(pid), memoryIntervalMS);
 
 const shouldMonitor = async () => enabled && permittedOS.includes(await currentOSPromise);
-const monitorEnabled = shouldMonitor();
 
 ProcessMonitor.startMonitor = async (startPID, processInfo) => {
-	if (!(await monitorEnabled)) return;
+	if (!(await shouldMonitor())) return;
+	if (dataByRid[processInfo.model]) delete dataByRid[processInfo.model];
```

Comment:
Think we forgot to change it here

```suggestion
	if (dataByRid[processInfo.Rid]) delete dataByRid[processInfo.Rid];
```